lib/modules/versioning/composer/index.ts

```diff
@@ -158,6 +158,14 @@ function minSatisfyingVersion(
   );
 }
 
+function subset(sub: string, dom: string): boolean {
+  if (npm.subset) {
```

Comment:
Is this likely to ever return false? Or did you just code it defensively in case?

Reply:
eslint complains that this method might not exists if there is no check.
But we should never enter the return false path.

Actually, I'm not happy with returning false in case it can not do the computation (due to absence of nom.subset or whatever reason) as it's not a truthy information. I was thinking of updating the return type to `boolean | undefined` or `boolean | null`. WDYT ?

Reply:
Not sure that achieves any better outcome? Either way you need to swallow an unexpected outcome

Reply:
I was thinking of this while trying to implement subset method for python versioning.

This versioning delegates it to poetry versioning if the specified range supports it, and fallback to pep440 versioning otherwise.
The implementation for poetry is pretty trivial, but not for pep440 as it'll require adding support for subsets in https://github.com/renovatebot/pep440.

I was thinking of returning undefined in the case of pep440 fallback in the meantime to allow fallbacking to version matching during constraints filtering.

Reply:
Implemented.